### ISSUE TITLE
Ripple effect on Ornata V2

### ISSUE THREAD

**Issue by Cr1ms0med** [1kc/razer-macos#296]
The Ripple effect not working at all with the razer Ornata V2. Might be glitched but a fix would be awesome thnx.

**Comment by dhobi** [comment]
Hi, could you give us a bit more info about your system:
- macOS 10 or 11, other?
- Which Razer macOS version do you use?

When activating the Ripple effect the first time, the system (macOS) should ask you if Razer macOS should be allowed to listen to any keypresses. Has such message popped up for you?

**Comment by spectre-110** [comment]
I am also having this issue. I am on macOS 10.15.7. My razer version is on 1.88.20. I will try updating my computer and return to see if it works after. If not, then please let me know if you need any other information.

**Comment by spectre-110** [comment]
Just updated everything, the ripple effect is still not working.

**Comment by Cr1ms0med** [comment]
I'm using macOS 10 and using the latest version of razer macOS I'm not sure if there is an extra step to set up the ripple effect but the system did open security and privacy went on the privacy tab and down to accessibility and checked Razer MacOS let me now if any info is missing or wrong.

**Comment by dhobi** [comment]
Does the light of the keyboard go off instantly when you select ripple?

Can you try this step by step:
1. Open menu
2. Select static color (green)
3. Open menu again 
4. Select ripple (green)
5. Open menu again (small lag?)
6. Start typing

Is ripple active?

**Comment by spectre-110** [comment]
Just tried this, and I could unfortunately not get it to work. Any other suggestions? Sorry for the trouble

**Comment by Cr1ms0med** [comment]
Did the steps It lags for a sec but when I switch to the ripple effect, The static green effect stays active. This is also the case when I switch to another color. ( The light of the keyboard does not turn off when switched to the ripple effect. It remains in the static green effect.

**Comment by avigucci** [comment]
Ripple was glitchy for me at first, but then I restarted my computer. Restarting seems to have worked for me.

**Comment by spectre-110** [comment]
> Did the steps It lags for a sec but when I switch to the ripple effect, The static green effect stays active. This is also the case when I switch to another color. ( The light of the keyboard does not turn off when switched to the ripple effect. It remains in the static green effect).

This is the same issue I am having as well.

**Comment by dhobi** [comment]
> Did the steps It lags for a sec but when I switch to the ripple effect, The static green effect stays active. 

I've checked the code for Ornata V2 and compared it to openrazer. Seems like we implemented support for Ornata V2 before openrazer (different naming) leading to a different implementation. I'm actually surprised that effects on the Ornata V2 worked at all 😄  
Anyway, I tried to port the Ornata V2 code over once again from openrazer. PR #298 will hopefully fix it (fingers crossed).
